### PR TITLE
Support non-1 indices

### PR DESCRIPTION
**Pull request by timholy**
Similar changes might be desired in QuartzImageIO, CC @rsrock.

Tests expected to fail on 0.6 due to https://github.com/JuliaLang/julia/issues/21369, but if you run the tests a second time in the same session, it passes locally on 0.6.

### CODE REVIEW

File: src/ImageMagick.jl [hunk 1]
```diff
@@ -228,11 +228,23 @@ mapIM(x::Normed) = x
...
+    collect_indices(indsA::Tuple{Vararg{OneTo}}, A) =
+        copy!(Array{eltype(A)}(length.(indsA)), A)
+    function collect_indices(indsA, A)
+        B = Array{eltype(A)}(length.(indsA))
+        copy!(B, CartesianRange(indices(B)), A, CartesianRange(indsA))
+    end
+else
+    const collect1 = collect
+end
+
```
Comment (timholy) [inline src/ImageMagick.jl]: Probably worth CCing @tkelman here. I don't think https://github.com/JuliaLang/julia/pull/21257 can safely be backported to julia-0.5. I guess the question is whether Compat should define `collect1`?

Comment (timholy) [reply src/ImageMagick.jl]: https://github.com/JuliaLang/Compat.jl/pull/350

**Comment by timholy** [comment]
Single test failure is a little weird, but since this has been changing recently (https://github.com/JuliaLang/julia/issues/21369) I think I won't worry about it just yet. (In any event, it's quite a corner case...)
